Implement RetrieveFastaSeq, LoadSeqs, and GenerateRandomBkgSeqs

kyrafetter opened the pull request.

kyrafetter committed:
Update myutils.py

add comments to RetrieveFastaSeq

Commit:
add comments to RetrieveFastaSeq

kyrafetter committed:
Merge remote-tracking branch 'origin/main' into formatting-sequence-input

Commit:
finish RetrieveFastaSeq

Commit:
fit import syntax for myutils

kyrafetter committed:
finish LoadSeqs, add GenerateRandomBkgSeqs

add code to LoadSeqs (finished), add template for GenerateRandomBkgSeqs to generate a set of random sequences from the fasta file

kyrafetter committed:
complete GenerateRandomBkgSeqs

add code to generate a random set of background sequences from fasta reference

kyrafetter committed:
Remove redundant imports

added imports needed in myutils.py so removed from spyglass.py

kyrafetter committed:
Update spyglass.py

Added imports back--need because directly used in spyglass.py

maczx-dev commented on spyglass/myutils.py:
```diff
@@ -35,31 +38,72 @@ def ERROR(msg):
+		# get a random chromosome
+		chrom = np.random.choice(chrs, 1)
+		# get a random start position on chosen chromosome
+		start = random.randrange(1, len(fasta[chrom].seq))
```
Subtract len of seq from end bound

kyrafetter committed:
fix index error

subtract seqLen from length of chrom so that sequence does not run off the end of the crhom
